pkg/curatedpackages/packageclient.go

```diff
@@ -46,3 +59,74 @@ func convertBundleVersionToPackageVersion(bundleVersions []packagesv1.SourceVers
 	}
 	return versions
 }
+
+func (pc *PackageClient) GeneratePackages() ([]packagesv1.Package, error) {
+	packageMap := pc.getPackageNameToPackage()
+	var packages []packagesv1.Package
+	for _, p := range pc.packages {
+		bundlePackage := packageMap[strings.ToLower(p)]
+		if bundlePackage.Name == "" {
+			fmt.Println(fmt.Errorf("unknown package %s", p).Error())
+			continue
+		}
+		name := CustomName + strings.ToLower(bundlePackage.Name)
+		packages = append(packages, convertBundlePackageToPackage(bundlePackage, name, pc.bundle.APIVersion))
+	}
+	return packages, nil
+}
+
+func (pc *PackageClient) WritePackagesToFile(packages []packagesv1.Package, d string) error {
+	directory := filepath.Join(d, packageLocation)
+	if !validations.FileExists(directory) {
+		if err := os.Mkdir(directory, dirPermission); err != nil {
+			return fmt.Errorf("unable to create directory %s", directory)
+		}
+	}
+
+	for _, pack := range packages {
+		displayPackage := NewDisplayPackage(pack)
+		content, err := yaml.Marshal(displayPackage)
+		if err != nil {
+			fmt.Println(fmt.Errorf("unable to parse package %s %v", pack.Name, err).Error())
+			continue
+		}
+		err = writeToFile(directory, pack.Name, content)
+		if err != nil {
+			fmt.Println(err.Error())
+		}
+	}
+	return nil
+}
+
+func writeToFile(dir string, packageName string, content []byte) error {
+	file := filepath.Join(dir, packageName) + ".yaml"
+	if err := os.WriteFile(file, content, filePermission); err != nil {
+		return fmt.Errorf("unable to write to the file: %s %v", file, err)
+	}
+	return nil
+}
+
+func (pc *PackageClient) getPackageNameToPackage() map[string]packagesv1.BundlePackage {
+	pMap := make(map[string]packagesv1.BundlePackage)
+	for _, p := range pc.bundle.Spec.Packages {
+		pMap[strings.ToLower(p.Name)] = p
+	}
+	return pMap
+}
+
+func convertBundlePackageToPackage(bp packagesv1.BundlePackage, name string, apiVersion string) packagesv1.Package {
+	p := packagesv1.Package{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: constants.EksaPackagesName,
```

Comment:
I thought the user was allowed to select their own namespace?

Reply:
There is no support for that as of now. There was a discussion that we would like to enforce eksa-packages for security reasons. I have added you to the discussion.

Reply:
https://amzn-aws.slack.com/archives/C02EMRQAM3N/p1651087131545479?thread_ts=1648589664.078929&cid=C02EMRQAM3N

Thanks for the context.